lib/src/data/datasource/remote/pokemon_api_service.dart

```diff
@@ -38,14 +39,43 @@ class PokemonApiService {
             pokemonResponseJsonList[i].url,
           );
           var evolvesTo = await _getEvolvesTo(pokemon);
+          pokemonList.add(
+            pokemon.copyWith(evolvesTo: evolvesTo),
+          );
+        }
+        pokemonResponse.pokemonResults.addAll(pokemonList);
+      }
+    } catch (e) {}
+    return pokemonResponse;
+  }
 
+  Future<PokemonResponse?> fetchPokemonPage(String direction) async {
+    var pokemonList = <Pokemon>[];
+    PokemonResponse? pokemonResponse;
+    try {
+      final response = await client.get(
+        Uri.parse(direction),
+      );
+      if (response.statusCode == NumericConstants.statusCode) {
+        pokemonResponse = PokemonResponse.fromJson(
+          json.decode(
+            response.body,
+          ),
+        );
+        var pokemonResponseJsonList = pokemonResponse.results;
+        for (var i = 0; i < pokemonResponseJsonList.length; i++) {
+          var pokemon = await fetchPokemonByIdList(
+            pokemonResponseJsonList[i].url,
+          );
+          var evolvesTo = await _getEvolvesTo(pokemon);
           pokemonList.add(
             pokemon.copyWith(evolvesTo: evolvesTo),
           );
         }
+        pokemonResponse.pokemonResults.addAll(pokemonList);
       }
```

Comment:
fetchPokemonPage and fetchPokemonList manage a similiar logic, do you think that is a better approach to wrap this logic into a method?

Reply:
Fixed